src/richText.js

```diff
@@ -205,19 +206,27 @@ define([
                         output = makeBubble(form, xpath, attrs);
                     insertHtmlWithSpace($('<p>').append(output).html());
                 }
+                return wrapper;
             },
             insertOutput: function (xpath) {
                 if (options.isExpression) {
                     throw new Error("cannot insert output into expression editor");
                 }
                 insertHtmlWithSpace(bubbleOutputs(xpath, form));
+                return wrapper;
+            },
+            change: function () {
+                editor.fire("saveSnapshot");
```

Comment:
Why did you choose `saveSnapshot` instead of `change` ? https://github.com/dimagi/Vellum/pull/680